### ISSUE TITLE
Training leads to HomRef calls

### ISSUE THREAD

**Issue by jguhlin** [google/deepvariant#251]
Hello, I'm currently attempting to train DV on a non-model organism. I believe I have the training examples being made properly. For ~170 samples, I take 30 out randomly to use as validation and shuffle both sets by splitting into 4096 files, then repeating a few times (shuffling the iterator as well).

Training proceeds quickly on the Tesla GPU, but all stats end up being homozygous reference. I can't tell what is going wrong.

Here's my make_examples
```
  /opt/deepvariant/bin/make_examples \
      --mode=training \
      --use_ref_for_cram=true \
      --ref=${reference} \
      --examples ${accession}.ds0.with_labels.examples \
      --sample_name ${accession} \
      --reads ${cram} \
      --truth_variants=${accession}.vcf.gz \
      --confident_regions=${accession}.bed \
      --regions CM0XXXXXX.1 \
      --write_run_info

```

Shuffling code is similar to this, but repeated multiple time:
```
raw_dataset = tf.data.TFRecordDataset(inputs)
for raw_record in itershuffle(raw_dataset, 2000):
  example = tf.train.Example()
  example.ParseFromString(raw_record.numpy())
  writer = random.choice(out_fhs)
  writer.write(example.SerializeToString())
```

And training code is like this:
```
  /opt/deepvariant/bin/model_train \
    --dataset_config_pbtxt=${config_path} \
    --batch_size=256 \
    --train_dir=${training_dir} \
    --model_name="inception_v3" \
    --learning_rate=0.008 \
    --start_from_checkpoint=/opt/models/wgs/model.ckpt \
    --number_of_steps=50000 \
    --save_interval_secs 300
```

Here is the run info for just one sample's examples set (only a single chromosome, for testing purposes, from the .run_info.pbtxt file):

```
labeling_metrics {
  n_truth_variant_sites: 3469
  n_truth_variant_alleles: 3474
  n_candidate_variant_sites: 9778
  n_candidate_variant_alleles: 9943
  n_non_confident_candidate_variant_sites: 2219
  n_true_positive_sites: 3468
  n_true_positive_alleles: 3845
  n_false_negative_sites: 1
  n_false_negative_alleles: 1
  n_false_positive_sites: 6309
  n_false_positive_alleles: 6469
  n_inexact_position_matches: 1
  n_exact_position_matches: 3469
  n_exact_position_and_allele_matches: 3443
  n_exact_position_and_allele_and_genotype_matches: 3443
}
```

Training runs just fine, with loss starting at ~1.2 and dropping to 0.04. Batch size is relatively small (memory error on the GPU with any larger). Is it simply my patience or is something else going on? I can provide tensorboard stats as well, but taking any model and performing make_examples(calling) -> postprocess results in only refcalls.

Thanks, and let me know what other info I can provide.

Edit: Here is some of the output from model_eval
```
Saving dict for global step 0: Accuracy/All = 0.17285156, Accuracy/Indels = 0.078431375, Accuracy/SNPs = 0.19634147, F1/All = 0.39246467, F1/Het = 0.0, F1/HomRef = 0.39246467, F1/HomVar = 0.2947544, FNs/All = 0.0, FNs/Indels = 0.0, FNs/SNPs = 0.0, FPs/All = 774.0, FPs/Indels = 164.0, FPs/SNPs = 610.0, Precision/All = 0.24414062, Precision/Het = 0.0, Precision/HomRef = 0.24414062, Precision/HomVar = 0.17285156, Precision/Indels = 0.19607843, Precision/SNPs = 0.25609756, Recall/All = 1.0, Recall/Het = 0.0, Recall/HomRef = 1.0, Recall/HomVar = 1.0, Recall/Indels = 1.0, Recall/SNPs = 1.0, TNs/All = 0.0, TNs/Indels = 0.0, TNs/SNPs = 0.0, TPs/All = 250.0, TPs/Indels = 40.0, TPs/SNPs = 210.0, global_step = 0, loss = 1.3173751
I1209 06:57:08.677582 46912496317632 estimator.py:2039] Saving 'checkpoint_path' summary for global step 0: ./model_training/model.ckpt-0
I1209 06:57:08.678164 46912496317632 model_eval.py:251] Eval results: {'Precision/Indels': 0.19607843, 'TNs/Indels': 0.0, 'Recall/Het': 0.0, 'F1/All': 0.39246467, 'FPs/Indels': 164.0, 'Accuracy/SNPs': 0.19634147, 'Recall/Indels': 1.0, 'TNs/SNPs': 0.0, 'global_step': 0, 'TNs/All': 0.0, 'FNs/All': 0.0, 'Precision/HomRef': 0.24414062, 'F1/Het': 0.0, 'FPs/All': 774.0, 'TPs/All': 250.0, 'TPs/SNPs': 210.0, 'FNs/Indels': 0.0, 'Recall/HomRef': 1.0, 'F1/HomVar': 0.2947544, 'F1/HomRef': 0.39246467, 'Accuracy/Indels': 0.078431375, 'Precision/Het': 0.0, 'Precision/SNPs': 0.25609756, 'Recall/All': 1.0, 'Precision/HomVar': 0.17285156, 'loss': 1.3173751, 'FNs/SNPs': 0.0, 'Recall/HomVar': 1.0, 'TPs/Indels': 40.0, 'Recall/SNPs': 1.0, 'Accuracy/All': 0.17285156, 'FPs/SNPs': 610.0, 'Precision/All': 0.24414062}


I1209 07:22:13.038631 46912496317632 estimator.py:1979] Saving dict for global step 3123: Accuracy/All = 0.7578125, Accuracy/Indels = 0.8181818, Accuracy/SNPs = 0.7436251, F1/All = 0.0, F1/Het = 0.0, F1/HomRef = 0.0, F1/HomVar = 0.0, FNs/All = 434.0, FNs/Indels = 62.0, FNs/SNPs = 372.0, FPs/All = 0.0, FPs/Indels = 0.0, FPs/SNPs = 0.0, Precision/All = 0.0, Precision/Het = 0.0, Precision/HomRef = 0.0, Precision/HomVar = 0.0, Precision/Indels = 0.0, Precision/SNPs = 0.0, Recall/All = 0.0, Recall/Het = 0.0, Recall/HomRef = 0.0, Recall/HomVar = 0.0, Recall/Indels = 0.0, Recall/SNPs = 0.0, TNs/All = 1358.0, TNs/Indels = 279.0, TNs/SNPs = 1079.0, TPs/All = 0.0, TPs/Indels = 0.0, TPs/SNPs = 0.0, global_step = 3123, loss = 0.9064512
I1209 07:22:13.039522 46912496317632 estimator.py:2039] Saving 'checkpoint_path' summary for global step 3123: ./model_training/model.ckpt-3123
I1209 07:22:13.040343 46912496317632 model_eval.py:251] Eval results: {'Precision/Indels': 0.0, 'TNs/Indels': 279.0, 'Recall/Het': 0.0, 'F1/All': 0.0, 'FPs/Indels': 0.0, 'Accuracy/SNPs': 0.7436251, 'Recall/Indels': 0.0, 'TNs/SNPs': 1079.0, 'global_step': 3123, 'TNs/All': 1358.0, 'FNs/All': 434.0, 'Precision/HomRef': 0.0, 'F1/Het': 0.0, 'FPs/All': 0.0, 'TPs/All': 0.0, 'TPs/SNPs': 0.0, 'FNs/Indels': 62.0, 'Recall/HomRef': 0.0, 'F1/HomVar': 0.0, 'F1/HomRef': 0.0, 'Accuracy/Indels': 0.8181818, 'Precision/Het': 0.0, 'Precision/SNPs': 0.0, 'Recall/All': 0.0, 'Precision/HomVar': 0.0, 'loss': 0.9064512, 'FNs/SNPs': 372.0, 'Recall/HomVar': 0.0, 'TPs/Indels': 0.0, 'Recall/SNPs': 0.0, 'Accuracy/All': 0.7578125, 'FPs/SNPs': 0.0, 'Precision/All': 0.0}

```

**Comment by jguhlin** [comment]
Looks like I was impatient. Now that it's been going for several hours (And reduced learning rate) tensorboard is giving better results. TPs dropped to 0, and everything dropped to 0, but now it's back up. I thought starting with the pre-trained wgs model would let it just improve but I guess it had to re-learn. Planning to scale up to the entire genome next.

**Comment by gunjanbaid** [comment]
@jguhlin glad to hear that training curves look reasonable! I want to mention that DV is currently written to be a diploid variant caller. In case you are retraining with data from polyploid organisms, it is not yet clear how DeepVariant will perform.

I'll close this issue for now, but feel free to reopen if you have any other questions!

**Comment by jumpyknight** [comment]
I'm not sure if you are still bothered by this issue, but I have posted a similar one before and got perfectly solved. Hope it can help you, too!

[https://github.com/google/deepvariant/issues/185](url)
